Sources/SIL/Printer.swift

```diff
@@ -50,6 +50,12 @@ class Printer: CustomStringConvertible {
         print(suf)
     }
 
+    func print<S: Collection>(_ xs: S, _ fn: (S.Element) -> Void) {
```

Comment:
I've added this because the current behavior of `print` is weird: `print(collection, "") { print($0) }` is equivalent to this overload, while `print(collection) { print($0) }` will print `collection.description`, which includes e.g. `[` and `]` for arrays.